ui/src/criteria/attribute.tsx

```diff
@@ -81,17 +129,115 @@ function AttributeEdit(props: AttributeEditProps) {
   const hintDisplayName = (hint: tanagra.EnumHintValue) =>
     hint.displayName || "Unknown Value";
 
-  const enumHintValues = underlay.entities
+  const attributeFilterHint = underlay.entities
     .find((g) => g.name === underlay.primaryEntity)
-    ?.attributes?.find((attribute) => attribute.name === props.data.attribute)
-    ?.attributeFilterHint?.enumHint?.enumHintValues;
+    ?.attributes?.find(
+      (attribute) => attribute.name === props.data.attribute
+    )?.attributeFilterHint;
+
+  const year = new Date().getFullYear();
+
+  if (isValid(attributeFilterHint?.integerBoundsHint)) {
+    const integerBoundsHint = attributeFilterHint?.integerBoundsHint;
+
+    if (
+      integerBoundsHint?.max !== null &&
+      integerBoundsHint?.max !== undefined &&
+      integerBoundsHint?.min !== null &&
+      integerBoundsHint?.min !== undefined
+    ) {
+      const min = year - integerBoundsHint?.max;
+      const max = year - integerBoundsHint.min;
+
+      const [minValue, setMinValue] = React.useState<number>(min);
+      const [maxValue, setMaxValue] = React.useState<number>(max);
+
+      const updateValue = (newMin: number, newMax: number) => {
+        setMinValue(newMin);
+        setMaxValue(newMax);
+        props.dispatchFn(
+          produce(props.data, (data) => {
+            data.min = newMin;
+            data.max = newMax;
+          })
+        );
+      };
+
+      React.useEffect(() => updateValue(min, max), []);
+
+      const handleChange = (event: Event, newValue: number | number[]) => {
+        newValue = newValue as number[];
+        const [newMin, newMax] = newValue;
+        updateValue(newMin, newMax);
+      };
+
+      const handleInputChange = (
+        event: React.ChangeEvent<HTMLInputElement>
+      ) => {
+        const newMin =
```

Comment:
I think you need some bounds checking here to ensure that `max > min`, especially since that can happen temporarily while entering a multi-digit number. You should maybe also ensure `min >= hintMin` and `max <= hintMax`, but look at AoU and see how it handles these cases.